include/deal.II/lac/tensor_product_matrix.h

```diff
@@ -624,6 +630,21 @@ TensorProductMatrixSymmetricSumBase<dim, Number, n_rows_1d>::apply_inverse(
 }
 
 
+
+template <int dim, typename Number, int n_rows_1d>
+std::size_t
+TensorProductMatrixSymmetricSumBase<dim, Number, n_rows_1d>::
+  memory_consumption() const
+{
+  return MemoryConsumption::memory_consumption(mass_matrix) +
+         MemoryConsumption::memory_consumption(derivative_matrix) +
+         MemoryConsumption::memory_consumption(eigenvalues) +
+         MemoryConsumption::memory_consumption(eigenvectors) +
+         MemoryConsumption::memory_consumption(tmp_array);
```

Comment:
This class looks a bit too expensive if one needs to create one instance for each cell.